src/CMakeLists.txt

```diff
@@ -10,7 +10,21 @@ include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
 
 add_subdirectory(crypto)
 add_subdirectory(univalue)
+
+add_custom_target(generate_build_info
+  BYPRODUCTS ${PROJECT_BINARY_DIR}/src/obj/build.h
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_BINARY_DIR}/src/obj
+  COMMAND ${PROJECT_SOURCE_DIR}/share/genbuild.sh ${PROJECT_BINARY_DIR}/src/obj/build.h ${PROJECT_SOURCE_DIR}
+  COMMENT "Generating obj/build.h"
+  VERBATIM
+)
+
 add_subdirectory(util)
+add_dependencies(bitcoin_util generate_build_info)
```

Comment:
It's unfortunate that this is less specific than what we have in master: depending on the entire library rather than making the relationship explicit:
```
libbitcoin_util_a-clientversion.$(OBJEXT): obj/build.h
```

But I suppose this will work just fine.

Reply:
Thanks! [Reworked](https://github.com/hebasto/bitcoin/pull/51#issuecomment-1815013786).

Reply:
Oh nice :)